cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorNonParametrizedTest.java

```diff
@@ -148,4 +148,80 @@ public void testCustomLabelsAndAnnotations(VertxTestContext context) {
+                .endSpec()
+                .build();
+        
+        ResourceOperatorSupplier supplier = ResourceUtils.supplierWithMocks(false);
+        SecretOperator secretOps = supplier.secretOperations;
+
+        ArgumentCaptor<Secret> clusterCaCert = ArgumentCaptor.forClass(Secret.class);
+        ArgumentCaptor<Secret> clusterCaKey = ArgumentCaptor.forClass(Secret.class);
+        ArgumentCaptor<Secret> clientsCaCert = ArgumentCaptor.forClass(Secret.class);
+        ArgumentCaptor<Secret> clientsCaKey = ArgumentCaptor.forClass(Secret.class);
+        when(secretOps.reconcile(eq(NAMESPACE), eq(AbstractModel.clusterCaCertSecretName(NAME)), clusterCaCert.capture())).thenAnswer(i -> Future.succeededFuture(ReconcileResult.created(i.getArgument(0))));
+        when(secretOps.reconcile(eq(NAMESPACE), eq(AbstractModel.clusterCaKeySecretName(NAME)), clusterCaKey.capture())).thenAnswer(i -> Future.succeededFuture(ReconcileResult.created(i.getArgument(0))));
+        when(secretOps.reconcile(eq(NAMESPACE), eq(KafkaCluster.clientsCaCertSecretName(NAME)), clientsCaCert.capture())).thenAnswer(i -> Future.succeededFuture(ReconcileResult.created(i.getArgument(0))));
+        when(secretOps.reconcile(eq(NAMESPACE), eq(KafkaCluster.clientsCaKeySecretName(NAME)), clientsCaKey.capture())).thenAnswer(i -> Future.succeededFuture(ReconcileResult.created(i.getArgument(0))));
+
+        KafkaAssemblyOperator op = new KafkaAssemblyOperator(vertx, new PlatformFeaturesAvailability(false, KubernetesVersion.V1_9), certManager, passwordGenerator,
+                supplier, ResourceUtils.dummyClusterOperatorConfig(1L));
+        Reconciliation reconciliation = new Reconciliation("test-trigger", Kafka.RESOURCE_KIND, NAMESPACE, NAME);
+
+        Checkpoint async = context.checkpoint();
+        
+        op.new ReconciliationState(reconciliation, kafka).reconcileCas(() -> new Date())
+                .onComplete(context.succeeding(c -> context.verify(() -> {
+                    assertThat(clusterCaCert.getAllValues(), hasSize(1));
+                    assertThat(clusterCaKey.getAllValues(), hasSize(1));
+                    assertThat(clientsCaCert.getAllValues(), hasSize(1));
+                    assertThat(clientsCaKey.getAllValues(), hasSize(1));
+
+                    Secret clusterCaCertSecret = clusterCaCert.getValue();
+                    Secret clusterCaKeySecret = clusterCaKey.getValue();
+                    Secret clientsCaCertSecret = clientsCaCert.getValue();
+                    Secret clientsCaKeySecret = clientsCaKey.getValue();
+
+                    assertThat(clusterCaCertSecret.getMetadata().getOwnerReferences(), hasSize(1));
+                    assertThat(clusterCaCertSecret.getMetadata().getOwnerReferences().get(0), is(ownerReference));
+
+                    assertThat(clientsCaCertSecret.getMetadata().getOwnerReferences(), hasSize(0));
+
+                    async.flag();
+                })));
+        }
+    }
```

Comment:
I think this is a almost good test for the positive case. You should also add a case for the other case - that the owner reference is disabled - and check that it is indeed null. It should be easy to build on that from the